pkg/server/util.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+import (
+	"net/http"
+
+	kcpclienthelper "github.com/kcp-dev/apimachinery/pkg/client"
+	"github.com/kcp-dev/logicalcluster"
+
+	"k8s.io/client-go/rest"
+)
+
+// NewClusterConfig wraps an existing config's roundtripper
+// with a custom cluster aware roundtripper.
+func NewClusterConfig(cfg *rest.Config) *rest.Config {
```

Comment:
Made a helper here. An apt location for this would probably be `kcp-dev/apimachinery`? But leaving here for now.

Reply:
Yes, please move to apimachinery. But looks good!